src/App.test.js

```diff
@@ -1,9 +1,87 @@
-import React from 'react';
-import ReactDOM from 'react-dom';
-import App from './App';
-
-it('renders without crashing', () => {
-  const div = document.createElement('div');
-  ReactDOM.render(<App />, div);
-  ReactDOM.unmountComponentAtNode(div);
+const puppeteer = require('puppeteer');
+const faker = require('faker');
+
+const person = {
+  name: faker.name.firstName() + ' ' + faker.name.lastName(),
+  email: "c@lvin.com",
+  phone: faker.phone.phoneNumber(),
+  message: faker.random.words(),
+  password: "password"
+};
+
+
+
+let browser;
+let page;
+const width = 1920;
+const height = 1080;
+
+beforeAll(async () => {
+  browser = await puppeteer.launch({
+    headless: false,
+    slowMo: 80,
+    args: [`--window-size=${width},${height}`]
+  });
+  page = await browser.newPage();
+  await page.setViewport({ width, height });
 });
+afterAll(() => {
+  browser.close();
+});
+
+
+
+const appUrlBase = 'http://localhost:3000'
+const routes = {
+  public: {
+    login: `${appUrlBase}/login`,
+    noMatch: `${appUrlBase}/PageNotFound`,
+  },
+  private: {
+    dashboard: `${appUrlBase}/`,
+    registration: `${appUrlBase}/registration`,
+  },
+};
+
+describe('Signing In', () => {
+  test('sign-in title loads correctly', async () => {
+
+    await page.goto(routes.public.login);
+    await page.waitForSelector('.header');
+
+    const html = await page.$eval('.header', e => e.innerHTML);
+    expect(html).toBe('sign in');
+
+
+  }, 16000);
+  test('login form works correctly', async () => {
+    // take a png screenshot and saves it inside root directory
+    await page.screenshot({ path: "loginScreen.png" })
+    // uses faker to load data from object uptop
+    await page.goto(routes.public.login);
+    await page.click('.email')
+    await page.type('.email', person.email)
+    await page.click('.password')
+    await page.type('.password', person.password)
+    await page.click('.remember')
+    await page.click('.signIn')
+
+    //This is supposed to show Dashboard, but can't find selector of `Dashboard`
+    const html = await page.$eval('`Dashboard`', e => e.innerHTML);
+
+    expect(html).toBe('Dashboard')
```

Comment:
You are trying to check if the Dashboard has rendered (you needed a `.` in front of the class name to select for a class), but a simpler way to check if login was successful is to see if the [url](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pageurl) has changed. You'll need to [wait](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pagewaitforselectororfunctionortimeout-options-args) for a bit to ensure the request has time to go through to the backend (500 milliseconds seems to work), and then check to see if the URL matches the dashboard's url.

Reply:
Tired to set the dashboard class with  `.` but got "not found error". If you look at the dashboard it has a class of `<div className="``Dashboard``">` with the backticks. 

I've changed the test per your request to be 
`expect(window.location.pathname).toBe('/')`